modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -811,6 +813,53 @@ public int getObjectEntriesCount(long groupId, long objectDefinitionId) {
 		return objectEntryPersistence.countByG_ODI(groupId, objectDefinitionId);
 	}
 
+	@Transactional(propagation = Propagation.REQUIRES_NEW)
+	public long getObjectEntriesCount(
+			long groupId, ObjectDefinition objectDefinition,
+			Predicate predicate)
+		throws PortalException {
+
+		DynamicObjectDefinitionTable dynamicObjectDefinitionTable =
+			_getDynamicObjectDefinitionTable(
+				objectDefinition.getObjectDefinitionId());
+
+		DynamicObjectDefinitionTable extensionDynamicObjectDefinitionTable =
+			_getExtensionDynamicObjectDefinitionTable(
+				objectDefinition.getObjectDefinitionId());
+
+		JoinStep joinStep = DSLQueryFactoryUtil.countDistinct(
+			dynamicObjectDefinitionTable.getPrimaryKeyColumn()
+		).from(
+			dynamicObjectDefinitionTable
+		).innerJoinON(
+			extensionDynamicObjectDefinitionTable,
+			extensionDynamicObjectDefinitionTable.getPrimaryKeyColumn(
+			).eq(
+				dynamicObjectDefinitionTable.getPrimaryKeyColumn()
+			)
+		);
+
+		if (!StringUtil.equals(objectDefinition.getScope(), "site")) {
```

Comment:
We have three possible scopes ([Site](https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/modules/apps/object/object-service/src/main/java/com/liferay/object/internal/scope/SiteObjectScopeProviderImpl.java), [Depot](https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/modules/apps/object/object-service/src/main/java/com/liferay/object/internal/scope/DepotObjectScopeProviderImpl.java) and [Company](https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/modules/apps/object/object-service/src/main/java/com/liferay/object/internal/scope/CompanyInstanceObjectScopeProviderImpl.java)). when it is adding an ObjectEntry, [this](https://github.com/liferay/liferay-portal/blob/e8cd1472e8089a733d3f8fa57e16fb6552a238ff/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java#L3960-L3972) validation is performed. Only in the _COMPANY_ scope the ObjectEntry must have groupId = 0.
Given that, I believe that it is safe ignore the groupId only if the scope is _COMPANY_, since the Depot is another one that the groupId is relevant.

If it is your intention ignore the Depot scope in the query, ignore this comment.

Reply:
UPDATE: I investigate further and discover that the scope by Depot (aka Asset Library) was disabled: https://liferay.atlassian.net/browse/LPS-140043

So you can ignore the comment above